packages/react/src/primitives/Message/MessageDismiss.tsx

```diff
@@ -0,0 +1,70 @@
+import * as React from 'react';
+import classNames from 'classnames';
+
+import { Button } from '../Button';
+import { VisuallyHidden } from '../VisuallyHidden';
+import { IconClose, useIcons } from '../Icon/internal';
+import { ComponentClassNames, ComponentText } from '../shared/constants';
+import { useMessageContext } from './useMessageContext';
+import { isFunction } from '@aws-amplify/ui';
+
+import {
+  MessageDismissProps,
+  BaseMessageDismissProps,
+  ForwardRefPrimitive,
+  Primitive,
+} from '../types';
+
+const MessageDismissPrimitive: Primitive<MessageDismissProps, 'button'> = (
+  {
+    onDismiss,
+    dismissButtonLabel,
+    hasIcon = true,
+    children,
+    className,
+    ...rest
+  },
+  ref
+) => {
+  const { setDismissed } = useMessageContext();
+  const icons = useIcons('message');
+
+  const dismissMessage = React.useCallback(() => {
+    setDismissed(true);
+    if (isFunction(onDismiss)) {
+      onDismiss();
+    }
+  }, [setDismissed, onDismiss]);
+  return (
+    <Button
+      variation="link"
+      colorTheme="overlay"
+      className={classNames(ComponentClassNames.MessageDismiss, className)}
+      ref={ref}
+      onClick={() => dismissMessage()}
+      {...rest}
```

Comment:
I was thinking if the developer wanted to implement their own showing/hiding of the Message where it stays in the DOM (we completely unmount ours, like Alert).

Reply:
What happens when they pass an `onClick` that just records an analytic?

Reply:
In that case, using OnClick, it would not do anything besides record the analytics (the message would remain). You would need to record your event in the onDismiss that you pass instead if you still want the usual dismiss behavior. Does that seem weird or confusing? 

Should we not allow an onDismiss on Message.Dismiss and whatever the customer passes to onClick is what happens "on dismiss"? Basically...
```
// in Message.Dismiss
const dismissMessage = React.useCallback(() => {
  if (isFunction(onClick)) {
    onClick();
  }
  setDismissed(true);
}, [setDismissed, onClick]);
```
Would we need to call setDismissed(true) last? I might be overthinking that this is a case developers would want (circumventing our dismiss functionality when they want to show/hide Message using css or something instead but still wanting to use our button)